### PR TITLE
Misc glob supports

### PR DESCRIPTION
**Pull request by kasimebrahim**
Miscellaneous changes to support GlobNode for PutLink, Unification and ...

### CODE REVIEW

File: opencog/atoms/core/Variables.cc [hunk 1]
```diff
@@ -539,12 +539,12 @@ Handle FreeVariables::substitute_scoped(const Handle& term,
 		if (GLOB_NODE == h->get_type())
 		{
 			Handle glst(substitute_scoped(h, args, silent, index_map, quotation));
-			if (glst->is_node())
-				return glst;
-
 			changed = true;
-			for (const Handle& gl : glst->getOutgoingSet())
-				oset.emplace_back(gl);
+			if (glst->get_type() == LIST_LINK)
```
Comment (linas) [inline opencog/atoms/core/Variables.cc]: I believe that this is unwrapping the results from `wrap_glb_with_list()`, is that right? If so, can you add a comment saying this?

**Comment by linas** [comment]
Wow. Looks absolutely perfect .. clean, crisp, minimal, organized, with unit tests. I really like it!

File: opencog/atoms/core/Variables.cc (hunk 1 above)
Comment (linas) [reply opencog/atoms/core/Variables.cc]: I will add this comment myself, in just a moment.

**Comment by kasimebrahim** [comment]
Thanks!

**Pull request merged**